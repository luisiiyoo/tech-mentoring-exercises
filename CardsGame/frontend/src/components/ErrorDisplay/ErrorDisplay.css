@import url(https://fonts.googleapis.com/css?family=Roboto:900);
@import url(https://fonts.googleapis.com/css?family=Open + Sans);

.Error {
  align-items: center;
  display: flex;
  flex-direction: column;
  font-size: 25px;
  font-family: 'Open Sans', sans-serif;
  height: inherit;
  justify-content: center;
  text-align: center;
  width: inherit;
}

.Error-Message {
  color: var(--font-color-app);
  text-transform: capitalize;
  margin: 25px 0 20px 0;
}

.Error-Label span {
  position: relative;
  background: var(--color-app);
  color: var(--font-color-app);
  font-size: 300%;
  padding: 0 20px;
  border-radius: 5px;
  font-family: 'Roboto', sans-serif;
  font-weight: bolder;
  transition: all 0.5s;
  cursor: pointer;
}

.Error-Label span:hover {
  background: var(--dark-color-app);
  color: var(--font-color-app);
  -webkit-animation: jelly 0.5s;
  -moz-animation: jelly 0.5s;
  -ms-animation: jelly 0.5s;
  -o-animation: jelly 0.5s;
  animation: jelly 0.5s;
}
