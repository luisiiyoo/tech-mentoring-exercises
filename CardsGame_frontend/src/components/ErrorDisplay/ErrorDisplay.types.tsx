export interface ErrorDisplayProps {
  message: string;
  statusCode?: number;
}
