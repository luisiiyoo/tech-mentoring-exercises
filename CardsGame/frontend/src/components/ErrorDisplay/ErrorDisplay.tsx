import React from 'react';
import { ErrorDisplayProps } from './ErrorDisplay.types';
import './ErrorDisplay.css';

const ErrorDisplay: React.FC<ErrorDisplayProps> = ({ message, statusCode }) => {
  const statusCodeStr = statusCode || '';
  return (
    <div className="Error">
      <div className="Error-Label" data-testid="ErrorLabel">
        <span> {`Error ${statusCodeStr}`} </span>
      </div>
      <div className="Error-Message" data-testid="ErrorMessage">
        {message}
      </div>
    </div>
  );
};

export default ErrorDisplay;
