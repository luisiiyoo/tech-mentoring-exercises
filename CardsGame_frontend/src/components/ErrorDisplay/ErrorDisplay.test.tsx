import React from 'react';
import { cleanup, render } from '@testing-library/react';
import ErrorDisplay from '../ErrorDisplay';

describe('ErrorDisplay', () => {
  afterEach(cleanup);

  it('should render Error component with a given message', () => {
    const messageMock = 'My Error';
    const { asFragment, getByTestId } = render(
      <ErrorDisplay message={messageMock} />,
    );
    expect(getByTestId('ErrorLabel')).toHaveTextContent('Error');
    expect(getByTestId('ErrorMessage')).toHaveTextContent(`${messageMock}`);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Error component with a given message and status code', () => {
    const messageMock = 'My Error';
    const statusCodeMock = 599;
    const { asFragment, getByTestId } = render(
      <ErrorDisplay message={messageMock} statusCode={statusCodeMock} />,
    );
    expect(getByTestId('ErrorLabel')).toHaveTextContent(
      `Error ${statusCodeMock}`,
    );
    expect(getByTestId('ErrorMessage')).toHaveTextContent(`${messageMock}`);
    expect(asFragment()).toMatchSnapshot();
  });
});
