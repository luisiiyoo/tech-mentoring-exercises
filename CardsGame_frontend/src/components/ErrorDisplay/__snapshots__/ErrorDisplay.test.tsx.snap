// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay should render Error component with a given message 1`] = `
<DocumentFragment>
  <div
    class="Error"
  >
    <div
      class="Error-Label"
      data-testid="ErrorLabel"
    >
      <span>
         Error  
      </span>
    </div>
    <div
      class="Error-Message"
      data-testid="ErrorMessage"
    >
      My Error
    </div>
  </div>
</DocumentFragment>
`;

exports[`ErrorDisplay should render Error component with a given message and status code 1`] = `
<DocumentFragment>
  <div
    class="Error"
  >
    <div
      class="Error-Label"
      data-testid="ErrorLabel"
    >
      <span>
         Error 599 
      </span>
    </div>
    <div
      class="Error-Message"
      data-testid="ErrorMessage"
    >
      My Error
    </div>
  </div>
</DocumentFragment>
`;
